calendar_reboot.js

```diff
@@ -689,39 +691,114 @@ Calendar.prototype.removeActiveCalendar = function() {
 };
     
 Calendar.prototype.saveCalendar = function() {
-        //saves the calendar in localStorage, saves the calendarUniqueId object
+        //saves the calendar in localStorage, saves the allCalendarIds object
         
         var self = this;
         //store the state in localStorage
-        storeInLocalStorage(self.calendarState.uniqueId, self.calendarState);
+        storeInLocalStorage(self.state.uniqueId, self.state);
         
-        //put calendar in calendarUniqueIds and store it
-        calendarUniqueId[self.calendarState.uniqueId] = self.calendarState.calendarTitle;
-        storeInLocalStorage(masterKey, calendarUniqueId);
+        //put calendar in allCalendarIdss and store it
+        allCalendarIds[self.state.uniqueId] = self.state.title;
+        storeInLocalStorage(allCalendarIdsKey, allCalendarIds);
         
 };
     
 Calendar.prototype.deleteCalendar = function() {
     //deletes the calendar from localStorage, deletes the calendar from the
-    //calendarUniqueId object
+    //allCalendarIds object
     
     var self = this;
-    //delete the calendar from the calendarUniqueId object
-    delete calendarUniqueId[calendar.calendarState.uniqueId];
+    //delete the calendar from the allCalendarIds object
+    delete allCalendarIds[self.state.uniqueId];
     //save that change in localstorage
-    storeInLocalStorage(masterKey, calendarUniqueId);
+    storeInLocalStorage(allCalendarIdsKey, allCalendarIds);
     
     //remove the calendarstate from localStorage
-    removeFromLocalStorage(calendar.calendarState.uniqueId);
+    removeFromLocalStorage(self.state.uniqueId);
 
 }
     
 
 //dictionary of savedCalendars, calendar title: unique ID
 
-var calendarUniqueId = {
+var allCalendarIds = {
 };
 
+//Make a storage manager
+var LocalCalendarStorage = function(params) {
+    var self = this;
+    var prefix = params['storeId'] || "";
+    
+    //params.storeId
+    
+    var toKey = function(id) {
+        //make a key out of a uniqueId
+        
+        return prefix + "_" + id;
+        
+    };
+    
+    self.getAllCalendarIds = function() {
+        //gets the allCalendarIds object from storage
+        
+        return loadFromLocalStorage(toKey(allCalendarIdsKey));
+    };
+    
+    self.save = function(appObj) {
```

Comment:
Since we only have calendar objects and this is a `LocalCalendarStorage`, it's probably best to just refer to these objects as `calendar`.

Alternatively, you could call this thing a `LocalStore` or something, but I feel like it might be more abstract than it needs to be. It makes it harder to know what the functions are assuming about the incoming data.